### PR TITLE
Update price range filter facet to use range component

### PR DESCRIPTION
**Pull request by lucasecdb**
Price ranges should be using a separate query parameter, instead of the `rest` used for brands and additional categories. Also, the price range facet should be using a range slider component, instead of a list of the facets options.

[Workspace](https://lucas--storecomponents.myvtex.com/eletronicos/smartphones).

#### Issues

* [ ] Update README.md with entrypoint component props
* [x] Update price range filter to use a separate query parameter (priceRange)
* [x] Create slider component to replace price range

### CODE REVIEW

File: react/components/FilterOptionTemplate.js [hunk 1]
```diff
@@ -31,11 +34,22 @@ export default class FilterOptionTemplate extends Component {
...
+    return filters.map(children)
+  }
+
   render() {
-    const { selected, children, filters, title, collapsable } = this.props
+    const { selected, title, collapsable, children, filters } = this.props
     const { open } = this.state
 
-    if (!filters.length) {
+    // keep old behaviour
```
Comment (andreldsa) [inline react/components/FilterOptionTemplate.js]: Change to "Backward-compatible support"

File: react/components/Range.js [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    step: PropTypes.number,
+    disabled: PropTypes.bool,
+    initialValues: PropTypes.shape({
+      left: PropTypes.number,
+      right: PropTypes.number,
+    }),
+  }
+
+  static defaultProps = {
+    min: 0,
```
Comment (vitoriaHeliane) [inline react/components/Range.js]: Não seria melhor usar constates?

File: react/components/Range.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+            }}
+            onDragStart={this.handleDragStart}
+            position="left"
+            active={this.state.dragging === 'left'}
+            value={this.state.values.left}
+          />
+          <RangeCircle
+            style={{
+              transform: `translateX(-${right}px) translateX(100%)`,
+              right: 0,
```
Comment (vitoriaHeliane) [inline react/components/Range.js]: Por que não usa a classe ```right-0``` do tachyons?

File: react/components/Range.js [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+                height: 3,
+                left,
+                right,
+              }}
+            />
+          </div>
+          <RangeCircle
+            style={{
+              transform: `translateX(${left}px) translateX(-100%)`,
+              left: 0,
```
Comment (vitoriaHeliane) [inline react/components/Range.js]: Por que não usa a classe ```left-0``` do tachyons?

File: react/components/Range.js (hunk 2 above)
Comment (lucasecdb) [reply react/components/Range.js]: why?

File: react/components/Range.js [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+    this.updateLayout()
+  }
+
+  render() {
+    const { disabled } = this.props
+    const { left, right } = this.state.translate
+
+    return (
+      <div className="ph4">
+        <div className="vtex-range w-100 relative pv4" style={{ height: 48 }}>
```
Comment (ClaudivanFilho) [inline react/components/Range.js]: you can use h2 `height: 32px` of tachyons and set a padding higher, like pv7

File: react/components/RangeCircle.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+RangeCircle.defaultProps = {
+  active: false,
+  disabled: false,
+  value: 0,
+  className: '',
+}
+
+RangeCircle.propTypes = {
+  className: PropTypes.string,
```
Comment (ClaudivanFilho) [inline react/components/RangeCircle.js]: Doc :eyes:

File: react/components/Range.js (hunk 5 above)
Comment (lucasecdb) [reply react/components/Range.js]: It doesn't work, I need at least 48px of height for the range to not visually break.

**Comment by lucasecdb** [comment]
[Update price range filter facet to use range component](https://trello.com/c/35JXjpZf/26-update-price-range-filter-facet-to-use-range-component)

**Pull request merged**